### PR TITLE
zchronod stage one implementation

### PR DESCRIPTION
**Pull request by sglk123**
**zchronod network impl
**zchronod framework impl
**zchronod doc

### CODE REVIEW

File: zchronod/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[package]
+name = "chrono"
```
Comment (fshif) [inline zchronod/Cargo.toml]: zchronod

File: zchronod/doc/zchrono_framework.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+<br>
+
+### Section 2 Arch&&DataFlow
+
+### zchronod_archtiecture
+
+<br>
+
+![](image/zchronod_arch.png)
```
Comment (fshif) [inline zchronod/doc/zchrono_framework.md:23]: Make sure the names in the image consistent with the package names

File: zchronod/network/src/rpc.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+use proto::zchronod::{ZchronodRequest, ZchronodResp};
+
+pub(crate) struct RpcServer {
+    port: String,
+}
+
+impl RpcServer {
+    pub(crate) fn new(listen: &str) -> Self {
+        RpcServer {
+            port: listen.to_string(),
```
Comment (fshif) [inline zchronod/network/src/rpc.rs]: confused arg "listen",what it means? ipaddr or Port? need more clear defination

File: zchronod/network/src/rpc.rs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+#[derive(Default)]
+pub struct ZchronodService {}
+
+#[tonic::async_trait]
+impl Zchronod for ZchronodService {
+    async fn send(&self, request: Request<ZchronodRequest>) -> Result<Response<ZchronodResp>, Status> {
+        info!("recv request from {:?}",request.get_ref().msg.as_ref().unwrap().id);
+        if let Some(mut ctx) = unsafe { CONTEXT.take() } {
+            ctx.get_network().send(request.get_ref().msg.clone().unwrap());
```
Comment (fshif) [inline zchronod/network/src/rpc.rs:37]: need more validation check

**Review by fshif — CHANGES_REQUESTED** [review]

File: zchronod/Cargo.toml (hunk 1 above)
Comment (sglk123) [reply zchronod/Cargo.toml]: changed

File: zchronod/doc/zchrono_framework.md (hunk 2 above)
Comment (sglk123) [reply zchronod/doc/zchrono_framework.md:23]: consistent with zchronod

File: zchronod/network/src/rpc.rs (hunk 3 above)
Comment (sglk123) [reply zchronod/network/src/rpc.rs]: change as self_address

File: zchronod/network/src/rpc.rs (hunk 4 above)
Comment (sglk123) [reply zchronod/network/src/rpc.rs:37]: this has not completed, the event received from rpc should pass to consensus layer which is vlc to encapsulate msg, and then send to gossip

**Review by fshif — APPROVED** [review]

**Pull request merged**